### ISSUE TITLE
Bug: @typescript-eslint/no-unused-vars false positive on catch clause argument type

### ISSUE THREAD

**Issue by MikeDabrowski** [typescript-eslint/typescript-eslint#10247]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.5.2&showAST=es&fileType=.ts&code=JYWwDg9gTgLgBAbzjAnmApnAggdwM4CiUU0cAvnAGYkhwDkAhvgLR4AmA1swMYA2w6AHYxmICNw50AUFID0suAElKcFBACucdYO4QQIIfBgALTACN0vCDjj9B6PHGbJTcdMVI5gvXnAtwAcwh7OQVdQTwjY2BBAIAVYwYYAHVoDjwALmx8IhIoOABeRDJQuHC8CF50ADorAIAKExj4xJS0vABKAG4ZGCgURCk4MuCKqtqIBroYBxg6bqkS7iTuYzh69CzcQg8oDsHhkxIbdEWgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgwBtSImjQO0SABpM4bDjKjYAcwAW%2BAKLzFkqKPjxI6iAF8AuutshbQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYwAviNUA&tokens=false

### Repro Code

```TypeScript
import { type AwsError } from 'aws-sdk-client-mock'

// If you uncomment the below lines - the error will be gone
// const thingThatWorks: AwsError = {}
// console.log(thingThatWorks);

try {
  console.log('test');
}
catch (e: AwsError) {
  throw e
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  "rules": {
    "@typescript-eslint/no-unused-vars": ["error", {
      "caughtErrors": "all"
    }]
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {}
}
```

### Expected Result

Expected no errors about AwsError being unused because it is used as a type in catch block

### Actual Result

'AwsError' is defined but never used. 1:15 - 1:23

**Comment by bradzacher** [comment]
This is working as expected - you cannot specify a type for the catch clause annotation unless it is one of `any` or `unknown`.

Your code has the following TS error:

> Catch clause variable type annotation must be 'any' or 'unknown' if specified.

[TS playground](https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzjAnmApnAggdwM4CiUU0cAvnAGYkhwDkAhvgLR4AmA1swMYA2w6AHYxmICNw50AUFID0suAElKcFBACucdYO4QQIIfBgALTACN0vCDjj9B6PHGbJTcdMVI5gvXnAtwAcwh7OQVdQTwjY2BBAIAVYwYYAHVoDjwALmx8IhIoOABeRDJQuHC8CF50ADorAIAKExj4xJS0vABKAG4ZGCgURCk4MuCKqtqIBroYBxg6bqkS7iTuYzh69CzcQg8oDsHhkxIbdEWpIA)

**Comment by MikeDabrowski-EQM** [comment]
Ok thank you for taking a look
